packages/common/src/content.ts

```diff
@@ -885,6 +891,66 @@ const getSolutionUrl = (content: IHubContent): string => {
   return hubUrl;
 };
 
+/**
+ * Create a new content with updated boundary properties
+ * @param content original content
+ * @param boundary boundary provenance
+ * @returns
+ */
+export const setContentBoundary = (
+  content: IHubContent,
+  boundary: GeographyProvenance
+) => {
+  // update content's item and boundary
+  const properties = { ...(content.item.properties || {}), boundary };
+  const item = { ...content.item, properties };
+  const updated = { ...content, item };
+  return { ...updated, boundary: getContentBoundary(updated) };
+};
+
+/**
+ * Create a new content with updated extent and derived properties like boundary
+ * @param content original content
+ * @param extent new extent
+ * @returns a new content with the updated extent and boundary
+ */
+export const setContentExtent = (
```

Comment:
Naming might be confusing, because it obtains a (non-extent) boundary and sets it as well.

Reply:
boundary is a property that depends on extent. and we don't have computed properties b/c this is a POJO. The way we decided to deal w/ this is to have setters for the lower level properties that derived properties depend on (in this case `extent`). Then those setters also update the derived properties (in this case `boundary`).